lib/api/dsl/index.js

```diff
@@ -11,90 +32,90 @@ const
+    return this.transformer.check(filters);
+  }
 
-/**
- * Checks if the provided filters are valid
- *
- * @param {object} filters
- * @return {Promise<Object>}
- */
-RealtimeEngine.prototype.validate = function validate (filters) {
-  return this.transformer.check(filters);
-};
+  /**
+   * Subscribes a filter to the real-time engine
+   *
+   * @param {string} index
+   * @param {string} collection
+   * @param {object} filters
+   * @return {Promise}
+   */
+  register(index, collection, filters) {
+    return this.transformer.normalize(filters)
+      .then(normalized => this.storage.store(index, collection, normalized));
+  }
 
-/**
- * Subscribes a filter to the real-time engine
- *
- * @param {string} index
- * @param {string} collection
- * @param {object} filters
- * @return {Promise}
- */
-RealtimeEngine.prototype.register = function register (index, collection, filters) {
-  return this.transformer.normalize(filters)
-    .then(normalized => this.storage.store(index, collection, normalized));
-};
+  /**
+   * Check if there are filters registered on an index-collection pair
+   *
+   * @param index
+   * @param collection
+   * @returns {boolean}
+   */
+  exists(index, collection) {
+    return this.storage.filtersIndex[index] !== undefined && this.storage.filtersIndex[index][collection] !== undefined;
+  }
 
-/**
- * Check if there are filters registered on an index-collection pair
- *
- * @param index
- * @param collection
- * @returns {boolean}
- */
-RealtimeEngine.prototype.exists = function exists (index, collection) {
-  return this.storage.filtersIndex[index] !== undefined && this.storage.filtersIndex[index][collection] !== undefined;
-};
+  /**
+   * Returns all filters IDs registered on an index-collection pair
+   *
+   * @param index
+   * @param collection
+   * @returns {Array} Array of matching filter IDs
+   */
+  getFilterIds(index, collection) {
+    return this.exists(index, collection) ? this.storage.filtersIndex[index][collection] : [];
+  }
 
-/**
- * Returns all filters IDs registered on an index-collection pair
- *
- * @param index
- * @param collection
- * @returns {Array} Array of matching filter IDs
- */
-RealtimeEngine.prototype.getFilterIds = function getFilterIds (index, collection) {
-  return this.exists(index, collection) ? this.storage.filtersIndex[index][collection] : [];
-};
+  /**
+   * Test data against filters in the filters tree to get the matching
+   * filters ID, if any
+   *
+   * @param {string} index - the index on which the data apply
+   * @param {string} collection - the collection on which the data apply
+   * @param {object} data to test filters on
+   * @param {string} [documentId] - if the data refers to a document, the document unique ID
+   * @return {Array} list of matching rooms
+   */
+  test(index, collection, data, documentId) {
+    if (this.exists(index, collection)) {
+      return this.matcher.match(index, collection, flattenObject(data, documentId));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 84 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)